fedimint-server/src/config/api.rs

```diff
@@ -1,775 +1,226 @@
-    requested_params: Option<ConfigGenParamsRequest>,
-    /// Our status
-    status: ServerStatus,
-    /// Configs that have been generated
-    config: Option<ServerConfig>,
-}
-
-/// Our local connection info
-#[derive(Debug, Clone)]
-struct ConfigGenLocalConnection {
-    /// Our TLS private key
-    tls_private: rustls::PrivateKey,
-    /// Our TLS public cert
-    tls_cert: rustls::Certificate,
-    /// Our guardian name
-    our_name: String,
-    /// URL of "leader" guardian to send our connection info to
-    /// Will be `None` if we are the leader
-    leader_api_url: Option<SafeUrl>,
-}
-
-impl ConfigGenState {
-    fn new(settings: ConfigGenSettings) -> Self {
-        Self {
-            settings,
-            auth: None,
-            local: None,
-            peers: BTreeMap::new(),
-            requested_params: None,
-            status: ServerStatus::AwaitingPassword,
-            config: None,
-        }
-    }
-
-    fn set_request(&mut self, request: ConfigGenConnectionsRequest) -> ApiResult<()> {
-        let (tls_cert, tls_private) = gen_cert_and_key(&request.our_name)
-            .map_err(|_| ApiError::server_error("Unable to generate TLS keys".to_string()))?;
-        self.local = Some(ConfigGenLocalConnection {
-            tls_private,
-            tls_cert,
-            our_name: request.our_name,
-            leader_api_url: request.leader_api_url,
-        });
-        info!(
-            target: fedimint_logging::LOG_NET_PEER_DKG,
-            "Set local connection for config gen"
-        );
         Ok(())
     }
 
-    fn local_connection(&self) -> ApiResult<ConfigGenLocalConnection> {
-        self.local.clone().ok_or(ApiError::bad_request(
-            "Our connection info not set yet".to_string(),
-        ))
-    }
+    pub async fn start_dkg(&self) -> anyhow::Result<()> {
+        let mut state = self.state.lock().await.clone();
 
-    fn auth(&self) -> ApiResult<ApiAuth> {
-        self.auth
+        let local_params = state
+            .local_params
             .clone()
-            .ok_or(ApiError::bad_request("Missing auth".to_string()))
-    }
+            .expect("The endpoint is authenticated.");
 
-    fn our_peer_info(&self) -> ApiResult<PeerServerParams> {
-        let local = self.local_connection()?;
-        Ok(PeerServerParams {
-            cert: local.tls_cert.clone(),
+        let our_peer_info = PeerConnectionInfo {
+            cert: local_params.tls_cert.0,
             p2p_url: self.settings.p2p_url.clone(),
             api_url: self.settings.api_url.clone(),
-            name: local.our_name,
-            status: Some(self.status.clone()),
-        })
-    }
+            name: local_params.name,
+            federation_name: local_params.federation_name,
+        };
 
-    fn get_peer_info(&self) -> BTreeMap<PeerId, PeerServerParams> {
-        self.peers
+        state
+            .connection_info
+            .insert(our_peer_info.api_url.clone(), our_peer_info);
```

Comment:
Same here, probably worth asserting that we didn't sneak in our own URL before (I know you have a check against that in the `add_peer_…` endpoint, but asserts are basically free)